### PR TITLE
Define the quota aspect interface.

### PR DESCRIPTION
**Pull request by geeknoid**
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/147)

### CODE REVIEW

File: pkg/aspect/quota/quota.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	// quota backends.
+	Adapter interface {
+		aspect.Adapter
+
+		// NewAspect returns a new quota implementation, based on the
+		// supplied Aspect configuration for the backend.
+		NewAspect(env aspect.Env, config proto.Message) (Aspect, error)
+	}
+
+    QuotaArgs struct {
```
Comment (douglas-reid) [inline pkg/aspect/quota/quota.go]: probably want some documentation on the struct itself.

File: pkg/registry/registrar.go [hunk 2]
```diff
@@ -26,4 +27,7 @@ type Registrar interface {
 
 	// RegisterDeny
 	RegisterDeny(denyChecker.Adapter) error
+
+	// RequestQuota
+	RegisterQuota(quota.Adapter) error
```
Comment (douglas-reid) [inline pkg/registry/registrar.go]: can we add more of useful comment here?

**Comment by geeknoid** [comment]
Updated. PTAL.

File: pkg/aspect/quota/quota.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	Adapter interface {
+		aspect.Adapter
+
+		// NewAspect returns a new quota implementation, based on the
+		// supplied Aspect configuration for the backend.
+		NewAspect(env aspect.Env, config proto.Message) (Aspect, error)
+	}
+
+	// QuotaArgs supplies the arguments for quota operations.
+	QuotaArgs struct {
```
Comment (douglas-reid) [inline pkg/aspect/quota/quota.go]: s/QuotaArgs/Args/ (reduce stutter of quota.QuotaArgs => quota.Args).

File: pkg/aspect/quota/quota.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+		// DeduplicationId is used for deduplicating quota allocation/free calls in the case of
+		// failed RPCs and retries. This should be a UUID per call, where the same
+		// UUID is used for retries of the same quota allocation or release call.
+		DeduplicationId string
```
Comment (douglas-reid) [inline pkg/aspect/quota/quota.go]: s/Id/ID/ (golang convention).

**Comment by geeknoid** [comment]
Done. PTAL.

**Pull request merged**